storm-client/src/jvm/org/apache/storm/security/auth/ThriftServer.java

```diff
@@ -45,34 +48,50 @@ public ThriftServer(Map storm_conf, TProcessor processor, ThriftConnectionType t
         } catch (Exception x) {
             LOG.error(x.getMessage(), x);
         }
+        try {
+            //locate our thrift transport plugin
+            ITransportPlugin transportPlugin = AuthUtils.GetTransportPlugin(_type, _storm_conf, _login_conf);
+            //server
+            _server = transportPlugin.getServer(_processor);
+            _port = transportPlugin.getPort();
+        } catch (IOException | TTransportException ex) {
+            handleServerException(ex);
+        }
+
     }
 
     public void stop() {
-        if (_server != null)
-            _server.stop();
+        _server.stop();
```

Comment:
I checked, AuthUtils.GetTransportPlugin can't return null, and neither included implementation of ITransportPlugin can return null :)